yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -391,6 +393,46 @@ func (d *JsonQueryFunctionDetector) GetIssues() []QueryIssue {
 	return issues
 }
 
+type UniqueNullsNotDistinctDetector struct {
+	query    string
+	detected bool
+}
+
+func NewUniqueNullsNotDistinctDetector(query string) *UniqueNullsNotDistinctDetector {
+	return &UniqueNullsNotDistinctDetector{
+		query: query,
+	}
+}
+
+// Detect checks if a unique constraint is defined which has nulls not distinct
+func (d *UniqueNullsNotDistinctDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_INDEX_STMT_NODE {
+		proto := msg.Interface().(proto.Message)
```

Comment:
Can you define a method in queryparser similar to `ProtoAsSelectStmt` that gets the struct itself. 
Also, when you are doing type assertions, always prefer using `x, ok := ` syntax to avoid panics. if `!ok`, return an error

Reply:
Yep done

Reply:
Haven't created a function for the PG_QUERY_TABLECONSTRAINT_NODE code below since it doesn't seem like a regularly used pattern. Do you think otherwise?

Reply:
I think the goal here is to keep most of the pure parser/proto related code in queryparser pkg. So even if not very re-usable, would recommend defining a function there. We also avoid importing and directly using pg_query pkg in queryissue package that way.

Reply:
Done